src/network/network_coordinator.cpp

```diff
@@ -156,6 +198,45 @@ bool ClientNetworkCoordinatorSocketHandler::Receive_SERVER_LISTING(Packet *p)
 	return true;
 }
 
+bool ClientNetworkCoordinatorSocketHandler::Receive_SERVER_CONNECTING(Packet *p)
+{
+	std::string token = p->Recv_string(NETWORK_TOKEN_LENGTH);
+	std::string join_key = p->Recv_string(NETWORK_JOIN_KEY_LENGTH);
+
+	/* Find the connecter based on the join-key. */
+	auto connecter_it = this->connecter_pre.find(join_key);
+	assert(connecter_it != this->connecter_pre.end());
```

Comment:
I'm not happy with the location of this assert, and any other in a similar location. Given it comes from the network the content of `Packet` should be considered invalid, i.e. it will trigger the assertion (or not in a release build, resulting in undefined behavior). So, it should either just ignore the Packet if it is with some comments or kill the connection with some error message.